### ISSUE TITLE
Add support for Lumen.

### ISSUE THREAD

**Issue by stevebaros** [mtvbrianking/laravel-mtn-momo#31]
After adding the library and manually registering the provider 
in production, it fires 500 indicating missing files.

My question is, does it fully support lumen or i need to get my hands a little more dirty ?

**Comment by mtvbrianking** [comment]
Actually the package doesn't support Lumen out of the box.

If you so wish, we can collaborate on porting it for Lumen.

**Comment by stevebaros** [comment]
![Screenshot_9](https://user-images.githubusercontent.com/11651953/70394792-5fa68d80-1a09-11ea-9d9e-fecd6e86aac2.png)

**Comment by stevebaros** [comment]
Let's port it for lumen, are you going to create a new branch or i can work straight from my fork master branch.

**Comment by mtvbrianking** [comment]
I've created a separate branch [`lumen-support`](https://github.com/mtvbrianking/laravel-mtn-momo/tree/lumen-support) for testing Lumen specifically.

You switch to it by making the following change in your `composer.json` file.

```diff
"require": {
    "php": "^7.2",
-   "bmatovu/multi-auth": "^6.0"
+   "bmatovu/multi-auth": "dev-lumen-support",
    "laravel/lumen-framework": "^6.0"
}
```
Regarding the `environmentFilePath()`; it doesn't exist on Lumen application instance so I've switched to a custom helper function to mirror the same desired result.

See commit: [Add env file path helper.](https://github.com/mtvbrianking/laravel-mtn-momo/commit/0f9f1870c122daf66c224e6af0515089852d29d7)

**Comment by stevebaros** [comment]
i think you meant 

> I've created a separate branch [`lumen-support`](https://github.com/mtvbrianking/laravel-mtn-momo/tree/lumen-support) for testing Lumen specifically.
> 
> You switch to it by making the following change in your `composer.json` file.
> 
> ```diff
> "require": {
>     "php": "^7.2",
> -   "bmatovu/multi-auth": "^6.0"
> +   "bmatovu/multi-auth": "dev-lumen-support",
>     "laravel/lumen-framework": "^6.0"
> }
> ```
> 
> Regarding the `environmentFilePath()`; it doesn't exist on Lumen application instance so I've switched to a custom helper function to mirror the same desired result.
> 
> See commit: [Add env file path helper.](https://github.com/mtvbrianking/laravel-mtn-momo/commit/0f9f1870c122daf66c224e6af0515089852d29d7)

Am thinking you meant 

`"require": {
        "php": "^7.2",
        "ext-json": "*",
        "bmatovu/laravel-mtn-momo": "dev-lumen-support",
        "darkaonline/swagger-lume": "^6.0",`

**Comment by mtvbrianking** [comment]
Do the intermediate changes fix the apparent Lumen issues.

**Comment by mtvbrianking** [comment]
I shall be closing this due to inactivity.

**Comment by stevebaros** [comment]
-- Well from my Tests in development everything seems to work fine. 
-- I will give a conclusive comment when I run it in production.

Nonetheless, you can merge it into master

**Comment by mtvbrianking** [comment]
Great, I'll do so and make an accompanying release.

**Comment by mtvbrianking** [comment]
Published release [v2.0.3](https://github.com/mtvbrianking/laravel-mtn-momo/releases/tag/v2.0.3) with this fix.
